src/librustc/sir.rs

```diff
@@ -200,4 +213,25 @@ impl SirFuncCx {
             _ => Err(()),
         }
     }
+
+    fn lower_binop(
+        &self,
+        op: mir::BinOp,
+        opnd1: &mir::Operand<'_>,
+        opnd2: &mir::Operand<'_>,
+        checked: bool,
+    ) -> ykpack::Rvalue {
+        let sir_op = binop_lowerings!(
+            op, Add, Sub, Mul, Div, Rem, BitXor, BitAnd, BitOr, Shl, Shr, Eq, Lt, Le, Ne, Ge, Gt,
+            Offset
+        );
+        let sir_opnd1 = self.lower_operand(opnd1);
+        let sir_opnd2 = self.lower_operand(opnd2);
+
+        if checked {
+            ykpack::Rvalue::CheckedBinaryOp(sir_op, sir_opnd1, sir_opnd2)
```

Comment:
FYI only `Add`, `Sub`, `Mul`, `Shl` and `Shr` can be checked.

Reply:
87f32fd adds an assertion. Do you think that's good enough?